File: AngularCarRental/BestCarsRental/src/app/shared/models/car.model.ts
import { Branch } from './branch.model';
import { CarType } from './car-type.model';

    export class Car {
        CarNumber: string;
        BestCondition: boolean;
        Mileage: number;
        Image: string;
		
		Branch: Branch = new Branch();
		CarType: CarType = new CarType();
    }

File: WebApi/BestCarsRental_BO/CarModel.cs

using System.ComponentModel.DataAnnotations;

namespace BestCarsRental_BO
{
    public class CarModel
    {    
        [Required]
        public int Mileage { get; set; }

        [Required, MinLength(2), MaxLength(50)]
        public string Image { get; set; }

        [Required]
        public int BestCondition { get; set; }

        [Required, MinLength(6), MaxLength(12)]
        public string CarNumber { get; set; }

        [Required]
        public BranchModel Branch { get; set; }

        [Required]
        public CarTypeModel CarType { get; set; }
    }
}
